contrib/nsqd.go

```diff
@@ -0,0 +1,40 @@
+package contrib
+
+import (
+	"github.com/nsqio/nsq/nsqd"
+)
+
+
+type INSQDAddon interface {
+	Enabled() bool
+	Start()
+}
+
+type NSQDAddons struct {
+	addons []INSQDAddon
+}
+
+// Starts all addons that are active
+func (as *NSQDAddons) Start() {
+
+	for _, addon := range as.addons {
+		addon.Start()
+	}
+}
+
+// Initializes addons that have options set
+func NewEnabledNSQDAddons(contribOpts []string, n *nsqd.NSQD) *NSQDAddons {
+	var activeAddons []INSQDAddon
+
+	n.Logf(nsqd.LOG_INFO, "Addons Initializing")
+
+	// ask each addon if it should be initialize
+	dogStats := NewNSQDDogStatsd(contribOpts, n)
+	if dogStats.Enabled() {
```

Comment:
No, I don't think it's a strict requirement. Just that it makes some sense to put this code which filters the mod-opts in one place, instead of in each module.

This also enables better "invalid option" checking - if the common code ensures that every option is for a module it knows of, and every module ensures that it understands every option it gets.